lib/Parse/ParseStmt.cpp

```diff
@@ -637,6 +641,12 @@ ParserResult<Stmt> Parser::parseStmt() {
     if (LabelInfo) diagnose(LabelInfo.Loc, diag::invalid_label_on_stmt);
     if (tryLoc.isValid()) diagnose(tryLoc, diag::try_on_stmt, Tok.getText());
     return parseStmtPoundAssert();
+  case tok::l_brace:
+    if (tryLoc.isValid()) diagnose(tryLoc, diag::try_on_stmt, Tok.getText());
+    SourceLoc colonLoc = Lexer::getTokenAtLocation(SourceMgr, PreviousLoc).getLoc();
+    diagnose(colonLoc, diag::labeled_block_needs_do)
+      .fixItInsert(colonLoc, "do");
```

Comment:
This insert `do` before the `:`. I believe `fixItInsert(Tok.getLoc(), "do ")` will do the job. And I think diagnostics location should be `{` loc too. So we don't need `colonLoc` at all.

Reply:
Sure. Updated.